app/controllers/taggings_controller.rb

```diff
@@ -14,6 +14,22 @@ def find_by_slug
     end
   end
 
+  def lookup_urls
+    content_lookup = ContentLookupForm.new(base_path: params[:base_path])
+
+    if content_lookup.valid?
+      content_item = ContentItem.find!(content_lookup.content_id)
+
+      render json: {
+        base_path: content_item.base_path,
+        content_id: content_item.content_id,
+        title: content_item.title
+      }
+    else
+      render json: { errors: content_lookup.errors }, status: 404
+    end
+  end
+
```

Comment:
I'm still not sure about the relationship between these two objects in

```ruby
    publisher = Tagging::TaggingUpdatePublisher.new(content_item, params[:tagging_tagging_update_form])

    if publisher.save_to_publishing_api
      redirect_to :back, success: "Tags have been updated!"
    else
      @tagging_update = Tagging::TaggingUpdateForm.from_content_item(content_item)
      @tagging_update.related_item_errors = publisher.errors
      @tagging_update.update_attributes_from_form(params[:tagging_tagging_update_form])
```

- It seems like it assumes that the publisher's errors are all to do with related links, which is technically true but could easily change
- All the data extracted from `#from_content_item` is overriden by `#update_attributes_from_form` except for `links`, which is used to expand base paths later on. This feels a bit weird to me as I'd expect to pass the request params to the form at initialisation time.
- `#title_for_related_link` doesn't work when you submit an invalid form with valid (but new) related items in it

Some naming things
- There's redundancy in `Tagging::TaggingUpdateForm` and `Tagging::TaggingUpdatePublisher`
- TaggingUpdateForm is still named form but it's separate from `app/forms`